source/common/runtime/runtime_impl.cc

```diff
@@ -454,9 +454,14 @@ void RtdsSubscription::onConfigUpdate(const std::vector<Config::DecodedResourceR
 
 void RtdsSubscription::onConfigUpdate(
     const std::vector<Config::DecodedResourceRef>& added_resources,
-    const Protobuf::RepeatedPtrField<std::string>&, const std::string&) {
-  validateUpdateSize(added_resources.size());
-  onConfigUpdate(added_resources, added_resources[0].get().version());
+    const Protobuf::RepeatedPtrField<std::string>& removed_resources, const std::string&) {
+  validateUpdateSize(added_resources.size(), removed_resources.size());
+
+  if (!added_resources.empty()) {
+    onConfigUpdate(added_resources, added_resources[0].get().version());
+  } else {
+    onConfigRemoved(removed_resources);
```

Comment:
Maybe comment that this is a singleton subscription, so we can only have the subscribed resource added or removed, but not both.

Reply:
👍 , added

Reply:
👍 , added

Reply:
Comment is good, but should we also have some ASSERTs, or handling of the case where adds and removes are delivered in one go?

Reply:
hi antonio

nice to meet you over github comment

i assume your suggestion is meant to have something handling the case when there are both adds and removes, there are a few lines of changes made in the `validateUpdateSize` method(starting from L477), would these changes cover your concern?